pkg/controllers/teammembership/teammembership_updater_controller.go

```diff
@@ -0,0 +1,250 @@
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	var team = new(greenhousev1alpha1.Team)
+	if err := r.Get(ctx, req.NamespacedName, team); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	var teamMembership = new(greenhousev1alpha1.TeamMembership)
+	err := r.Get(ctx, types.NamespacedName{Name: team.Name, Namespace: team.Namespace}, teamMembership)
+	if !apierrors.IsNotFound(err) && err != nil {
+		return ctrl.Result{}, err
+	}
+
+	teamMembershipExists := !apierrors.IsNotFound(err)
+
+	if team.Spec.MappedIDPGroup == "" {
+		if teamMembershipExists {
+			membersCountMetric.With(prometheus.Labels{
+				"namespace": team.Namespace,
+				"team":      team.Name,
+			}).Set(float64(0))
+
+			log.FromContext(ctx).Info("deleting TeamMembership, Team does not have MappedIdpGroup set", "team-membership", teamMembership.Name)
+			err = r.Delete(ctx, teamMembership, &client.DeleteOptions{})
+			return ctrl.Result{}, err
+		} else {
+			log.FromContext(ctx).Info("Team does not have MappedIdpGroup set", "team", team.Name)
+			return ctrl.Result{}, nil
+		}
+	}
+
+	teamMembershipStatus := initTeamMembershipStatus(teamMembership)
+	defer func() {
+		// Set status only if TM exists.
+		if teamMembership.Name != "" {
+			if statusErr := r.setStatus(ctx, teamMembership, teamMembershipStatus); statusErr != nil {
+				log.FromContext(ctx).Error(statusErr, "failed to set status")
+			}
+		}
+	}()
+
+	// Ignore organizations without SCIM configuration.
+	if organization.Spec.Authentication == nil || organization.Spec.Authentication.SCIMConfig == nil {
+		log.FromContext(ctx).Info("SCIM config is missing from org", "Name", req.NamespacedName)
+
+		c := greenhousev1alpha1.FalseCondition(greenhousev1alpha1.ScimAccessReadyCondition, greenhousev1alpha1.SecretNotFoundReason, "SCIM config is missing from organization")
+		teamMembershipStatus.SetConditions(c)
+
+		return ctrl.Result{}, nil
+	}
+
+	scimClient, err := r.createScimClient(ctx, req.Namespace, &teamMembershipStatus, organization.Spec.Authentication.SCIMConfig)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	users, err := r.getUsersFromScim(scimClient, team.Spec.MappedIDPGroup)
+	if err != nil {
+		log.FromContext(ctx).Info("failed processing team-membership for team", "error", err)
+		teamMembershipStatus.SetConditions(greenhousev1alpha1.FalseCondition(greenhousev1alpha1.ScimAccessReadyCondition, greenhousev1alpha1.ScimRequestFailedReason, ""))
+		return ctrl.Result{}, err
+	}
+
+	teamMembershipStatus.SetConditions(greenhousev1alpha1.TrueCondition(greenhousev1alpha1.ScimAccessReadyCondition, "", ""))
+
+	membersCountMetric.With(prometheus.Labels{
+		"namespace": team.Namespace,
+		"team":      team.Name,
+	}).Set(float64(len(users)))
+
+	if teamMembershipExists && slices.Equal(teamMembership.Spec.Members, users) {
+		// Requeue when nothing has changed.
+		return ctrl.Result{RequeueAfter: TeamMembershipRequeueInterval}, nil
+	}
```

Comment:
This is covered by the `clientutil.CreateOrPatch(..)`.  It will fetch the TeamMembership, copy it, apply the mutate(..) and only if there is a diff it will perform the patch.